rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -294,7 +294,7 @@ public synchronized int joinLRA(StringBuilder recoveryUrl, URI lra, long timeLim
             timeLimit = 0;
         }
 
-        if (!transaction.isActive()) {
+        if (!transaction.isActive() && !transaction.isRecovering()) {
```

Comment:
This code needs some documentation since, unless the reader is aware of the change you made to `Transaction#enlistParticipant`, he will assume that inactive participants are allowed to attempt to join with the LRA.